packages/ra-ui-materialui/src/defaultTheme.ts

```diff
@@ -22,22 +22,49 @@ export const defaultTheme = {
         closedWidth: 50,
     },
     components: {
+        MuiAutocomplete: {
+            variants: [
+                {
+                    props: {},
+                    style: ({ theme }) => ({
+                        [theme.breakpoints.down('sm')]: {
+                            // @see MuiTextField.styleOverrides.root
+                            width: '100%',
+                        },
+                    }),
+                },
+            ],
+        },
         MuiFilledInput: {
-            styleOverrides: {
-                root: {
-                    backgroundColor: 'rgba(0, 0, 0, 0.04)',
-                    '&$disabled': {
+            variants: [
+                {
+                    props: {},
+                    style: {
                         backgroundColor: 'rgba(0, 0, 0, 0.04)',
+                        '&$disabled': {
+                            backgroundColor: 'rgba(0, 0, 0, 0.04)',
+                        },
                     },
                 },
-            },
+            ],
```

Comment:
Can you explain why you needed to change these lines too?

Reply:
Please see: https://github.com/marmelab/react-admin/pull/8797#issuecomment-1496479298

> As the usage of styleOverrides looks not recommended for the future, I also took the initiative to migrate the existing on a separated commit: https://github.com/marmelab/react-admin/commit/b217bb621e94097bcfe38a58ee75c26585092353

If you prefer to keep the change to the target, I can simply drop this commit. :+1: